TM1py/Objects/Cube.py

```diff
@@ -112,6 +112,72 @@ def _construct_body(self) -> str:
         body_as_dict['Dimensions@odata.bind'] = [format_url("Dimensions('{}')", dimension)
                                                  for dimension
                                                  in self.dimensions]
-        if self.has_rules:
+        if str(self.rules):
             body_as_dict['Rules'] = str(self.rules)
         return json.dumps(body_as_dict, ensure_ascii=False)
+
+    def enable_rules(self):
+        if not self.rules.text:
+            # If there is no rule, there is nothing to do.
+            return
+
+        rules_statements = self.rules.text.splitlines()
+        if not len(rules_statements) == 1:
+            raise RuntimeError(
+                "The cube rules are not disabled correctly. "
+                "Must be 1 line of base64-encoded hash.")
+
+        encoded_rules_statement = rules_statements[0]
+        if not encoded_rules_statement.startswith(RULES_ENCODING_PREFIX):
+            raise RuntimeError(
+                f"The cube rules are not disabled correctly. "
+                f"Must start with prefix: '{RULES_ENCODING_PREFIX}'")
+
+        encoded_rule = encoded_rules_statement[len(RULES_ENCODING_PREFIX):]
+        self._rules = Rules(base64.b64decode(encoded_rule).decode('utf-8'))
+
+    def enable_feeders(self):
+        if not self.rules:
+            # If there is no rule, there is nothing to do.
+            return
+
+        rules_statements = self.rules.text.splitlines()
+
+        encoded_feeders_statement = rules_statements[-1]
+        if not encoded_feeders_statement.startswith(FEEDERS_ENCODING_PREFIX):
+            raise RuntimeError(
+                f"The cube feeders are not disabled correctly. "
+                f"First line in Feeders section must start with prefix: '{FEEDERS_ENCODING_PREFIX}'")
+
+        encoded_feeders = encoded_feeders_statement[len(FEEDERS_ENCODING_PREFIX):]
+        self._rules = Rules(
+            self.rules.text[:-len(encoded_feeders_statement)] + base64.b64decode(encoded_feeders).decode('utf-8'))
+
+    def disable_feeders(self):
+        if not self.rules:
```

Comment:
Consider adding a RunTimeError when attempting to disable feeders in a rule without feeders

Reply:
As a TM1py user, I would prefer not to receive an error in that scenario. If we did raise it, I would have to handle it on each write function just in case that feeders might be removed at some point in the cube.

But that's just my opinion

Reply:
Just pushed another commit. Neither empty rules nor empty feeders will raise an error.

I think that's the most convenient from a user perspective. Especially if we introduce this as an optional argument on the write function.